plugin/CactbotEventSource/FateWatcher.cs

```diff
@@ -132,26 +124,96 @@ public struct CEDirectorData {
     private static Dictionary<int, int> fates;
     private static Dictionary<int, CEDirectorData> ces;
 
-    public FateWatcher(CactbotEventSource client, string language) {
+    private static string GetRemoteOpcodeJSON(string url) {
+      string json = null;
+      using (HttpClient http = new HttpClient()) {
+        var opcodeTask = http.GetStringAsync(url);
+        json = opcodeTask.GetAwaiter().GetResult();
+      }
+      return json;
+    }
+
+    public FateWatcher(CactbotEventSource client, CactbotEventSourceConfig Config, string language) {
       client_ = client;
+      CEOpcodeList localOpcodeList = new CEOpcodeList();
+
+      if (!Config.DisableAutomaticOpcodeUpdates) {
```

Comment:
This is a pretty long function.  Is it possible to have a `TryToGetRemoteOpcodes()` function that returns null or the remote opcodes to wrap some of this?